api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupMER_TX_ML2019Report.java

```diff
@@ -0,0 +1,214 @@
+        CohortDefinition diedOfCancer = df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("116030AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")), BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition diedOfInfectiousDiseases = df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("73d67c86-06df-4863-9819-ccb2a6bb98f8")), BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition diedOfNonInfectiousDiseases = df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("13a7b84b-b661-48a5-8315-0bcc0174e5c8")), BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition diedOfNaturalCauses = df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("84899c95-d455-4293-be6f-7db600af058f")), BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition diedOfNonNaturalCauses = df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("ab115b14-8a9f-4185-9deb-79c214dc1063")), BaseObsCohortDefinition.TimeModifier.LAST);
+        CohortDefinition diedOfUnknown =df.getPatientsWithCodedObsDuringPeriod(getCauseOfDeathConcept(),null, Arrays.asList(hivMetadata.getConcept("dcd6865a-30ab-102d-86b0-7a5022ba4115")), BaseObsCohortDefinition.TimeModifier.LAST);
+
+        addAgeAndGender( dsd, "a", "died", df.getPatientsInAll(TX_ML,diedTraceOutcome));
+        addAgeAndGender( dsd, "b", "silentlyTransferred", df.getPatientsInAll(TX_ML,silentlyTransferred));
+        addAgeAndGender( dsd, "c", "LTFP L3MONTHS",  df.getPatientsInAll(TX_ML,havingArtStartDateDuringQuarter));
+        addAgeAndGender( dsd, "d", "LTFP G3MONTHS", df.getPatientsInAll(TX_ML,havingArtStartDateBeforeQuarter));
+        addAgeAndGender( dsd, "e", "stoppedARVTaking",  df.getPatientsInAll(TX_ML,ClientsStoppedonART));
+        addAgeAndGender( dsd, "f", "died Of TB",  df.getPatientsInAll(TX_ML,diedOfTB));
+        addAgeAndGender( dsd, "g", "died Of Cancer",  df.getPatientsInAll(TX_ML,diedOfCancer));
+        addAgeAndGender( dsd, "h", " other died Of Infectious Diseases", df.getPatientsInAll(TX_ML, diedOfInfectiousDiseases));
+        addAgeAndGender( dsd, "i", "died Of non infectious diseases",  df.getPatientsInAll(TX_ML,diedOfNonInfectiousDiseases));
+        addAgeAndGender( dsd, "j", "died Of Natural causes",  df.getPatientsInAll(TX_ML,diedOfNaturalCauses));
+        addAgeAndGender( dsd, "k", "died Of non natural causes",  df.getPatientsInAll(TX_ML,diedOfNonNaturalCauses));
+        addAgeAndGender( dsd, "l", "unknown cause of death",  df.getPatientsInAll(TX_ML,diedOfUnknown));
+
+        return rd;
+    }
+
+    public void addAgeAndGender(CohortIndicatorDataSetDefinition dsd, String key, String label, CohortDefinition cohortDefinition) {
```

Comment:
@dbaluku for such general methods, can we also make them general such that they can be reused by other reports?

Reply:
@solemabrothers this is template dependent and so it can  change depending on the excel template, i advise we keep it here since template always change